### ISSUE TITLE
Custom notificator implementation example (Pushjet)

### ISSUE THREAD

**Issue by Abyss777** [traccar/traccar#3953]
We have refactored notifications and here is example how easy to integrate new own transports now.
Used https://pushjet.io/ as example.

The code is here
https://github.com/Abyss777/traccar/commit/49dcc0e71978a72aba271ee353f66754408f42fa

Example configuration
```xml
    <entry key='notificator.types'>web,mail,pushjet</entry>
    <entry key='notificator.pushjet.class'>org.traccar.notification.NotificationPushjet</entry>
    <entry key='notificator.pushjet.url'>https://api.pushjet.io/</entry>
```

User have to have `pushjet.service` attribute with service secret token (registered manually). Client must be subscribed to this service to get notifications. This can be automated from the client application side.

PS: the issue can be closed and used as an example. Also I can send a PR if need to include it in main project.

**Comment by tananaev** [comment]
Let me try to implement proper push notifications first, but let's keep this here for other people who want to use push notification now.

**Comment by Abyss777** [comment]
It is not about pushes itself, just how implement notifications via some web service api. It can be SMS or telegram or any other.

**Comment by tananaev** [comment]
Makes sense. I think we should implement some SMS notifications alternatives (API instead of SMPP) in future as well. This is one of the most requested things.

**Comment by Abyss777** [comment]
I believe it is not a secret... but small note.
Just after adding
```
wrapper.java.classpath.1=.
```
to `conf/wrapper.conf`

Traccar (java) will lookup `.class` files in traccar folder to load in classpath.
It makes easier deploy and support own notificators without recompiling main jar.

**Comment by tananaev** [comment]
Interesting. Maybe we should have a tutorial on how to implement your own notificator, how to compile and configure it.

**Comment by JBFelizberto** [comment]
Thís is a good Idea. Hope that It Will can use on windows plataforma ( in future )as well

**Comment by deanet** [comment]
+ for telegram

**Comment by acastrovargas** [comment]
I think many users would be happy with "telegram" notifications. I currently use E2T.link but it is not ideal for demanding users.

![Image of Yaktocat](https://image.ibb.co/cFaMYz/1.png)

This Spanish telegram group of 394 users, would be happy with this

**Comment by Abyss777** [comment]
It won't be me who implement telegram support. I do not trust service which denies work without my phone number.

**Comment by acastrovargas** [comment]
you are right.

**Comment by oliv3** [comment]
Hi, is this merged ?

**Comment by tananaev** [comment]
No, it's not. We have our own push notifications now.

**Comment by oliv3** [comment]
In 4.0 ? I must have missed something, then. Any quick link ? Thanks !

**Comment by tananaev** [comment]
No, it will be available in the future release (517382ccc972d6d1ba2b5fd2d912cc6e5ac83902).

**Comment by oliv3** [comment]
Cool. I'll wait, then. No date set, I guess ?

**Comment by oliv3** [comment]
(I'm asking because I have an upcoming PR real soon for a platform-aware build script for the installers)

**Comment by tananaev** [comment]
No dates yet. Possibly in a month or so.

**Comment by Abhi931** [comment]
HI tananaev.. I am using traccar 3.16  and try to integrate SMS notification but after adding all the details on traccar.xml related to SMPP service. I am not able to get the SMS notification in my Mobile number.
Please find the information that I added in traccar.xml

	<entry key='notificator.types'>web,mail,sms</entry>
	<entry key='sms.enable'>true</entry>
	<entry key='notificator.sms.manager.class'>org.traccar.smpp.SmppClient</entry>
	<entry key='sms.smpp.enable'>true</entry>
	<entry key='filter.enable'>true</entry>
	<entry key='filter.zero'>true</entry>
	
	<entry key='sms.smpp.host'>******</entry>
	<entry key='sms.smpp.port'>7721</entry>
	<entry key='sms.smpp.username'>test</entry>
	<entry key='sms.smpp.password'>test007</entry>
	<entry key='sms.smpp.notificationsCharset'>GSM</entry>
	<entry key='sms.smpp.notificationsDataCoding'>0</entry>
	<entry key='sms.smpp.sourceAddress'>TECHS</entry>
	
Please tell me if I am going wrong and what I need to change.
Thanks in advance

**Comment by tananaev** [comment]
This is the place to discuss it.

**Comment by oliv3** [comment]
Priceless.

**Comment by oliv3** [comment]
How did my comment unfortunately disappeared ?

**Comment by andreapx** [comment]
> 
> 
> No dates yet. Possibly in a month or so.

Hi, I don't see this option yet, hasn't this feature been implemented?
And about the first post, I really haven't understood how to use pushjet. Where is the pushjet.service attribute and where should I put the secret token?
Thanks!!!
